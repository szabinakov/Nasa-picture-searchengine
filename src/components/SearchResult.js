import React from 'react'
import PropTypes from 'prop-types'
import '../styles/SearchResult.css'


const SearchResult = ({results}) => {
    const images = results.map((image) => image.links[0].href)

    const foundImages = images.map((image, index) => 
    <img key={index} className='imageOnScreen' src={image} alt='imageOfResult'></img>
    )

    if(!results.length) {
        return <p>No Results!</p>
    } else {
    return(
            <ul className='imageContainer'>
                <li>{foundImages}</li>
            </ul>
        
    )}

}

SearchResult.propTypes = {
    results: PropTypes.array
}
export default SearchResult 