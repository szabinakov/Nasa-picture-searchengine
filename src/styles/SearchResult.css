body {
    font-size: 30px;
    font-family: 'Courier New', Courier, monospace;
    background-color: rgba(192, 192, 192, 0.37);
}


.imageContainer {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    width: 100%;
}

.imageOnScreen {
    margin: 10px;
    margin-top: 100px;
    border: solid;
    width: 40%;
    height: 100%;
    border-width: 10px;
    border-radius: 30px;
    border-color: rgb(172, 6, 6);
}

.imageOnScreen:hover {
    border-color: rgb(255, 0, 0);
}
