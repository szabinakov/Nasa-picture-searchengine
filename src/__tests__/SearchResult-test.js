import React from 'react';
import { render } from '@testing-library/react';
import SearchResult from '../components/SearchResult';

describe('SearchResult', () => {
    const mockResults = ['firstpic', 'secondpic', 'thirdpic']

    const {asFragment} = render (<SearchResult results={mockResults}/>)

    it('it renders correctly', () => {
        
        expect(asFragment()).toMatchSnapshot()
    })
})