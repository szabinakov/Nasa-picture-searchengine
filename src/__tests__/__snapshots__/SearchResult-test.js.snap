// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult it renders correctly 1`] = `
<DocumentFragment>
  <div
    class="imageContainer"
  >
    <img
      alt="imageOfResult"
      class="imageOnScreen"
      src="firstpic"
    />
    <img
      alt="imageOfResult"
      class="imageOnScreen"
      src="secondpic"
    />
    <img
      alt="imageOfResult"
      class="imageOnScreen"
      src="thirdpic"
    />
  </div>
</DocumentFragment>
`;
